src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp

```diff
@@ -469,7 +467,7 @@ bool SmokeCoAlarmServer::GetAttribute(EndpointId endpointId, AttributeId attribu
     EmberAfStatus status = getFn(endpointId, &value);
     bool success         = (EMBER_ZCL_STATUS_SUCCESS == status);
 
-    if (!success)
+    if (!success && EMBER_ZCL_STATUS_UNSUPPORTED_ATTRIBUTE != status)
```

Comment:
if UNSUPPORTED is a possible result, how about changing signature of these methods to return an enum like `success/fail/not_supported` ? that way callers can keep failing on actual failures but ignore unsupported.

Reply:
Because modifying the return value requires more changes. So I added an mUnsupportedStatus variable to determine if there are unsupported attributes.